### DISCUSSION TITLE
CB1 and Waveshare RS485 CAN HAT for Raspberry Pi

### DISCUSSION THREAD

**Discussion by PeterHouseJr** [bigtreetech/CB1#34] [General]
Can the CB1 support the Waveshare RS485 CAN HAT?

Where is config.txt for device table installation of the Waveshare SPI driver?

Thank you.

**Comment by QuinFromCanada** [comment]
I too would love an answer to this.
At the moment I'm trying to stumble through this with mine new builds.

does the following line in the BoardEnv.txt apply to external SPI devices like the RS485 Can Hat from waveshare, or only BTT boards with internal CANBUS modules such as the Octopus?

`
#uncomment the following overlays for MCP2515 spi to canbus module

overlays=mcp2515
`
Right now I'm playing with this line in BoardEnv.txt:

`
#uncomment the following overlays and parameters to release 'spidev1.1' to use>
NOTE: 'spidev1.1' cannot be used together with TFT35_SPI and MCP2515

overlays=spi-spidev

param_spidev_spi_bus=1

param_spidev_spi_cs=1

param_spidev_max_freq=1000000
`

Which does seem to work in enableing the SPI however, I get a msg that MCU cant connect on reboot.  Rebooting the Klipper firmware from the web interface does fix it but its a bit annoying.

@PeterHouseJr I'll post here if I manage to find the answer, I've only been at it for a short time.

**Reply by AlfredBroda** [reply]
So, how did the experiments go, is this combination reliable? I don't have this CANHAT yet, I have a BTT U2C which works on USB. I would love to free a USB slot (and get rid of a cable) and use the hat option for CAN.

**Reply by PeterHouseJr** [reply]
I solved it by getting a Pi4.  The CB1 is an uphill battle all the way and the tech support is poor due to the relatively low number of users.

I back-ordered a Pi4 CM from both Digikey and Mouser and received one in about 6 weeks and then cancelled the other order.

All is now working well at this time.

**Reply by Bestsage** [reply]
Hello guys, I'm coming late but the main problem for you is that the spi pins on the cb1 aren't the same as the ones fo the pi4
